### PR TITLE
Tests and documentation

### PR DESCRIPTION
**Pull request by TheAwiteb**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -302,12 +321,21 @@ impl<K: KeyExtractor, M: RateLimitingMiddleware<QuantaInstant>> GovernorConfigBu
         }
     }
 
+    /// Set x-ratelimit headers to response, the headers is
+    /// - `x-ratelimit-limit`       - Request limit
+    /// - `x-ratelimit-remaining`   - The number of requests left for the time window
+    /// - `x-ratelimit-after`       - Number of seconds in which the API will become available after its rate limit has been exceeded
+    /// - `x-ratelimit-whitelisted` - If the request method not in methods, this header will be add it, use [`methods`] to add methods
+    ///
+    /// By default `x-ratelimit-after` is enable, with [`use_headers`] will enable `x-ratelimit-limit`, `x-ratelimit-whitelisted` and `x-ratelimit-remaining`
```
Comment (AaronErhardt) [inline src/lib.rs]: Typo: "is enabled"

File: examples/custom_key.rs [hunk 2]
```diff
@@ -1,81 +1,99 @@
...
-        .key_extractor(RealIpKeyExtractor)
+        .key_extractor(UserToken::default())
+        .use_headers()
         .finish()
         .unwrap();
 
     HttpServer::new(move || {
         App::new()
-            // Put the reverse proxy IP in app data so that Governor middleware can access it
-            .app_data(web::Data::new(trusted_reverse_proxy_ip))
```
Comment (AaronErhardt) [inline examples/custom_key.rs:78]: I don't think we should overwrite the existing example. It has a valid use case and should help everyone who want's to rate limit behind a reverse proxy. I think the old and the new one should both be available.

**Review by AaronErhardt — CHANGES_REQUESTED** [review]
Except for a few small things, it looks very good. We should be able to merge this very soon and publish a new release.

File: examples/custom_key.rs (hunk 2 above)
Comment (TheAwiteb) [reply examples/custom_key.rs:78]: Yes, I see that, I will return the old example with some modifications (moving the struct outside the main) and modify the new example (removing the token field)

**Comment by TheAwiteb** [comment]
Super force push 😁

**Review by AaronErhardt — APPROVED** [review]
Looks good to me, I will likely release a new version over the new two days.

**Comment by TheAwiteb** [comment]
It is better to release it after the next feature, which is the feature to change the response of the TooManyRequests error, and also bring the error about the inability to fetch the key from the user and be included in the KeyExtractor and this will cause a bracing change

**Comment by TheAwiteb** [comment]
Or if you want to include the headers in a separate release to avoid bracing change as much as possible, which you like

**Comment by AaronErhardt** [comment]
If there are more changes you would like to add over the next days I will wait with the release. We already have breaking changes even if those are minor so it definitely makes sense to wait.

**Pull request merged**